### ISSUE TITLE
Interactions browser: truncated download files are confusing

### ISSUE THREAD

**Issue by KatjaSchulz** [jhpoelen/eol-globi-data#169]
If I do a query on what taxa interact with tuna, 490 data rows are listed in the browser, representing interactions of 9 distinct taxa of tuna with 346 distinct target taxa:
http://www.globalbioticinteractions.org/browse/index.html#interactionType=interactsWith&resultType=json&sourceTaxon=Thunnus

When I download the data, I actually get a file with over 2000 records, but only 13 target taxa are represented.  I know this is due to a limit in the number of rows for the download file, but it's very confusing for the user.  We need to provide a lot more help in navigating the constraints of this interface.  Here are a couple of suggestions:

1. Explain why there may be more interactions in the download file.  
1.1. In the interactions table use "unique interactions" instead of just "interactions" when reporting the interactions count. 
1.2. Where you provide the download links, add the following text:
Download interactions data: csv json dot
The file may include multiple records for each unique interaction.

2. Don't give them truncated download files.
If the number of interactions exceeds what can be accommodated in the download file, ask the user to add constraints to their query.  Where you would show the download link, show a warning like this instead:  "Too many interactions to fit into a download file.  To enable data download, please try a more specific search."

**Comment by jhammock** [comment]
Also in case 2, perhaps an additional sentence:
"Too many interactions to fit into a download file. To enable data download, please try a more specific search. To access all GloBI records, please try the Data Access link."

**Comment by jhpoelen** [comment]
I very much agree that truncating data is not desirable. In fact, I'd rather have a system that simply retrieves the data requested, no matter how many data records are included. I can see some quick fixes: (a) disable download all together (a non-option I assume) (b) only include distinct interactions (loosing links to the references) (c) build a system that is able to stream results, no matter what size or (d) implement a method to count records satisfying a specific query, call this prior to enabling the download to assure that the number of records is lower than the download limit.

While (d) might be a little bit easier to implement, I prefer option (c) by simply returning all. 

I'll have to think a little about how to address this important issue.

@jhammock Perhaps another feature for GloBI III?

**Comment by jhpoelen** [comment]
PS I marked this issue as a bug-feature - one of those issues that could be interpreted either way, depending who you talk to.

**Comment by jhammock** [comment]
Yeah, let's queue this up. If there's an easy way to implement d as a stopgap, I'd favor doing that in the meantime, but if that's still significant work, let's skip it and make it c for GloBI III
